### ISSUE TITLE
[Woo POS] Improve handling of error when trying to pay for an already paid order

### ISSUE THREAD

**Issue by joshheald** [woocommerce/woocommerce-ios#13128]
> `paymentSuccess(done: () -> Void)`
>
> I'm not really sure what cancellation looks like here. It shouldn't be possible to go back in this case (though it is right now.)
> It shows an "Unable to process payment – this order is already paid" error, with a Cancel Payment button. The button resets the payment, but should probably do the same as New transaction. Otherwise, the user can press Collect Payment again and get the same error repeatedly.

_Originally posted by @joshheald in https://github.com/woocommerce/woocommerce-ios/pull/13100#pullrequestreview-2135255822_

The comment above isn't necessarily the best way to handle this... it will need some careful thought.

When we show the error that an order is already paid, we don't want the merchant to unthinkingly go back and charge the customer again... however, it's less likely in POS for that to happen, as the orders are always created in-line.

On the other hand, if something goes wrong and the merchant somehow ends up back editing the cart of an already paid order, and is trying to charge a _new_ customer for a new order instead, we need to give them a way to make a new order with a cleared cart.

Ideally we need to design this out as much as possible too!

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by jaclync** [comment]
Do you think trying to pay for an already paid order is still possible with the latest M1 design where there's no CTA in the success state to go back to update the cart and check out again?

![Image](https://github.com/woocommerce/woocommerce-ios/assets/1945542/947d4498-31f9-4311-86c2-a452c2699d32)

**Comment by joshheald** [comment]
Not in the happy path, but it's definitely still possible.

The main case will be an error on the response from the `capture_terminal_payment` request to WooPayments at the end of the flow. This could happen even if the capture was successful, especially in poor network conditions. 

In that case we should quietly refresh the order and see whether it's been paid, but if we get repeated errors from the order endpoint, the merchant could go back and try again. They might start a new transaction, which would make it hard to prevent double charges.

It would _probably_ fail to take the payment, because if the orders endpoint is facing issues, they're not going to get that far... but we need the merchant to understand the risks and that they need to be sure whether or not the merchant has been charged. @joe-keenan has this on his list (though I now realise I didn't see it in the error designs yesterday.)

**Comment by joshheald** [comment]
> Not in the happy path, but it's definitely still possible.
> 
> The main case will be an error on the response from the `capture_terminal_payment` request to WooPayments at the end of the flow. This could happen even if the capture was successful, especially in poor network conditions.
> 
> In that case we should quietly refresh the order and see whether it's been paid, but if we get repeated errors from the order endpoint, the merchant could go back and try again. They might start a new transaction, which would make it hard to prevent double charges.
> 
> It would _probably_ fail to take the payment, because if the orders endpoint is facing issues, they're not going to get that far... but we need the merchant to understand the risks and that they need to be sure whether or not the merchant has been charged. [@joe-keenan](https://github.com/joe-keenan) has this on his list (though I now realise I didn't see it in the error designs yesterday.)

Ignore all that. The above is important but it's a totally separate issue.

I think we can deprioritise the original issue since it's not really possible to get in that situation any more.

**Comment by joshheald** [comment]
Hah, I'm wrong again. Sorry for the confusion.

This error can happen if we get to the situation described above, can't refresh the order (or do, but think it's not paid for even though it is), and then the merchant taps `Try again`.

If they do that, it retries the payment and shows the "This order is already paid" error if it's paid.

In that case, we should go to a new transaction... but it probably shouldn't look much like an error, because the merchant may think that it's unpaid because of not reading the message.

We could show the `Payment successful` screen in this case instead... WDYT?

Let's do this after #13162

**Comment by jaclync** [comment]
> If they do that, it retries the payment and shows the "This order is already paid" error if it's paid.
> 
> In that case, we should go to a new transaction... but it probably shouldn't look much like an error, because the merchant may think that it's unpaid because of not reading the message.
> 
> We could show the `Payment successful` screen in this case instead... WDYT?

Yup this makes sense 👍
